src/canvas/PatternPath.tsx

```diff
@@ -9,6 +9,8 @@ export class PatternPath implements IPatternPath {
     private _path2D: Path2D;
     private _isPath2DValid: boolean;
     private _lastIndexAddedToPath2D: number;
+    private _isSelected: boolean;
```

Comment:
I think we should move isSelected and is Highlighted out to a different Selection class. I mentioned the reasons in my previous comment.